src/backend/distributed/executor/multi_router_executor.c

```diff
@@ -453,8 +458,7 @@ CitusModifyBeginScan(CustomScanState *node, EState *estate, int eflags)
 	 */
 	LockPartitionsInRelationList(distributedPlan->relationIdList, AccessShareLock);
 
-	/* modify tasks are always assigned using first-replica policy */
-	workerJob->taskList = FirstReplicaAssignTaskList(taskList);
+	workerJob->taskList = taskList;
```

Comment:
There should probably have been a comment, but changing this line breaks the rebalancer.